### PR TITLE
Shared object build

### PR DESCRIPTION
**Pull request by Shillaker**
Adding shared object builds of gRPC and Faabric

### CODE REVIEW

**Review by csegarragonz — APPROVED** [review]

**Comment by csegarragonz** [comment]
LGTM, but there seems to be a problem with the tests.

File: include/faabric/util/chaining.h [hunk 1]
```diff
@@ -4,10 +4,6 @@
 #include <proto/faabric.pb.h>
 
 namespace faabric::util {
-std::string postJsonFunctionCall(const std::string& host,
```
Comment (Shillaker) [inline include/faabric/util/chaining.h:7]: This isn't used anywhere

File: include/faabric/util/files.h [hunk 2]
```diff
@@ -4,9 +4,6 @@
 #include <string>
 #include <vector>
 
-#define FILE_PATH_HEADER "FilePath"
-#define EMPTY_FILE_RESPONSE "Empty response"
```
Comment (Shillaker) [inline include/faabric/util/files.h:8]: Split this into `files.h` and `http.h`

File: src/util/CMakeLists.txt [hunk 3]
```diff
@@ -34,9 +34,6 @@ add_dependencies(util rapidjson_ext)
 target_link_libraries(
         util
         proto
-        ssl
-        curl
-        crypto
```
Comment (Shillaker) [inline src/util/CMakeLists.txt:39]: These were to support libcurl usage

File: src/util/files.cpp [hunk 4]
```diff
@@ -67,64 +65,6 @@ size_t writeDataCallback(void* ptr, size_t size, size_t nmemb, void* stream)
...
-        throw FileNotFoundAtUrlException(msg);
-    }
-
-    if (out.str() == "IS_DIR") {
-        throw faabric::util::FileAtUrlIsDirectoryException(url +
-                                                           " is a directory");
-    }
-
-    return faabric::util::stringToBytes(out.str());
-}
```
Comment (Shillaker) [inline src/util/files.cpp:126]: This has been replaced with the implementations in `http.cpp`

**Review by csegarragonz — APPROVED** [review]

**Pull request merged**